src/Streamly/Internal/Data/Array/Foreign.hs

```diff
@@ -104,7 +104,7 @@ module Streamly.Internal.Data.Array.Foreign
     -- , writeChunksOfFrom
     -- , ...
 
-    , writeIndex
+    --, writeIndex
```

Comment:
Instead of commenting it out please move to Data.Array.Foreign.Mut module. Even the commented out writeSlice etc should be moved there.

Reply:
Yes, there needs to be a lot of learning. This is only a minor change for some correctness.